Patch borrowed from
http://cgit.openembedded.org/cgit.cgi/openembedded/tree/recipes/mysql/files/configure-ps-cache-check.patch

It allows to specify through ac_cv_FIND_PROC how ps should be used on
the target to find the PID of a program.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: mysql-4.1.22/configure.in
===================================================================
--- mysql-4.1.22.orig/configure.in	2009-01-28 16:33:28.000000000 +0000
+++ mysql-4.1.22/configure.in	2009-01-28 16:43:26.000000000 +0000
@@ -471,8 +471,8 @@
 
 # Lock for PS
 AC_PATH_PROG(PS, ps, ps)
-AC_MSG_CHECKING("how to check if pid exists")
-PS=$ac_cv_path_PS
+AC_CACHE_CHECK([how to check if pid exists], [ac_cv_FIND_PROC],
+[
 # Linux style
 if $PS p $$ 2> /dev/null | grep $0 > /dev/null
 then
@@ -511,8 +511,9 @@
       AC_MSG_ERROR([Could not find the right ps switches. Which OS is this ?. See the Installation chapter in the Reference Manual.])
   esac
 fi
-AC_SUBST(FIND_PROC)
-AC_MSG_RESULT("$FIND_PROC")
+ac_cv_FIND_PROC="$FIND_PROC"
+])
+AC_SUBST([FIND_PROC], [$ac_cv_FIND_PROC])
 
 # Check if a pid is valid
 AC_PATH_PROG(KILL, kill, kill)
